controllers/reconcile.go

```diff
@@ -853,6 +853,97 @@ func setupFedModules(feEnv *crd.FrontendEnvironment, frontendList *crd.FrontendL
 	return nil
 }
 
+func adjustSearchEntry(searchEntry *crd.SearchEntry, frontend crd.Frontend) crd.SearchEntry {
+	newSearchEntry := crd.SearchEntry{
+		// make the id environment and frontend specific to reduce duplicate ids across Frontend resources
+		ID:          fmt.Sprintf("%s-%s-%s", frontend.Name, frontend.Spec.EnvName, searchEntry.ID),
+		Title:       searchEntry.Title,
+		Description: searchEntry.Description,
+		Href:        searchEntry.Href,
+		AltTitle:    searchEntry.AltTitle,
+		IsExternal:  searchEntry.IsExternal,
+	}
+	return newSearchEntry
+}
+
+func setupSearchIndex(feList *crd.FrontendList) []crd.SearchEntry {
+	searchIndex := []crd.SearchEntry{}
+	for _, frontend := range feList.Items {
```

Comment:
Over the course of this change you switch between `felist` and `feList` - probably better to just go with one.